File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        return nums[nums.size()/2];
       // int key=0;
       //  int element=0;
       //  int n=nums.size();
       //  map<int,int>m;
       //  for(int i=0;i<n;i++)
       //  {
       //      m[nums[i]]++;
       //  }
       //  for(auto i=m.begin();i!=m.end();i++)
       //  {
       //      if(i->second>key)
       //      {
       //          key=i->second;
       //          element=i->first;
       //       }
       //  }
       //  return element;
    }
};


       
      
          


       

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element=0;
        int n=nums.length;
        Map<Integer,Integer> mp=new HashMap<Integer,Integer>();
        for(int num:nums)
        {
            if(!mp.containsKey(num))
            {
                mp.put(num,1);
            }
            else
                mp.put(num,mp.get(num)+1);
            if(mp.get(num)>nums.length/2)
            {
                element=num;
                break;
            }
        }
        return element;
    }
}

 
    
    